src/js/components/TeamList.js

```diff
@@ -0,0 +1,44 @@
+import { getTeams, addTeam } from "@lib/api";
+import { getEl, containsClass } from "@js/util";
+import { teamTemplate, teamAddBtnTemplate } from "@js/template";
+import { MESSAGES } from "@constants/constant";
+
+class TeamList {
+  constructor(store) {
+    this.store = store;
+    this.container = getEl(".team-list-container");
+    this.init();
+  }
+
+  async init() {
+    this.store.on("teamList", this.render.bind(this));
+    this.setTeamList();
+    this.container.addEventListener("click", this.clickDelegationHandler.bind(this));
+  }
+
+  async setTeamList() {
+    const { data: teamList } = await getTeams();
+    this.store.set({
+      teamList: [...teamList],
+    });
+  }
+
+  clickDelegationHandler({ target }) {
+    if (containsClass(target, "add-team-button")) return this._addTeamHandler();
+  }
+
+  async _addTeamHandler() {
+    const name = prompt(MESSAGES.ADD_TEAM);
+    if (name === null) return;
+    await addTeam(name);
+    this.setTeamList();
+  }
+
+  render() {
+    const { teamList } = this.store.get();
+    const template = teamList.map((team) => teamTemplate(team)).join("");
```

Comment:
```suggestion
const template = teamList.map(teamTemplate).join("");
```
이렇게 간략하게 작성하는 것도 가능합니다! ㅎㅎ

Reply:
헉 문화충격이네요 감사합니다